torch/distributed/rpc/backend_registry.py

```diff
@@ -171,21 +171,58 @@ def _tensorpipe_init_backend_handler(store, name, rank, world_size, rpc_backend_
     from . import TensorPipeAgent
 
     if not isinstance(store, dist.Store):
-        raise RuntimeError("`store` must be a c10d::Store. {}".format(store))
+        raise TypeError("`store` must be a c10d::Store. {}".format(store))
 
     if not isinstance(
         rpc_backend_options, TensorPipeRpcBackendOptions
     ):
-        raise RuntimeError(
+        raise TypeError(
             "`rpc_backend_options` must be a `TensorPipeRpcBackendOptions`. {}".format(
                 rpc_backend_options
             )
         )
 
-    agent = TensorPipeAgent(
-        store, name, rank, world_size, rpc_backend_options
+    # The agent's join method is required to behave like a barrier and perform
+    # collective operations, for which it relies on a process group, instead of
+    # re-implementing this on top of RPCs.
+
+    # Initialize ProcessGroup.
+    if dist.is_initialized():
+        raise RuntimeError(
+            "Default process group must not be initialized before init_rpc."
+        )
+
+    process_group_timeout = rpc_constants.DEFAULT_PROCESS_GROUP_TIMEOUT
+
+    dist.init_process_group(
```

Comment:
Let's land this one for now, and add an issue (similar to #33583) to avoid write the default global pg.

Reply:
I created #39138 to track this